src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -4425,7 +4425,7 @@ public void checkImportedPackagesObservable(final JCCompilationUnit toplevel) {
                 TypeSymbol tsym = imp.qualid.selected.type.tsym;
                 if (tsym.kind == PCK && tsym.members().isEmpty() &&
                     !(Feature.IMPORT_ON_DEMAND_OBSERVABLE_PACKAGES.allowedInSource(source) && tsym.exists())) {
-                    log.error(DiagnosticFlag.RESOLVE_ERROR, imp.pos, Errors.DoesntExist(tsym));
+                    log.error(DiagnosticFlag.RESOLVE_ERROR, imp.qualid.selected.pos, Errors.DoesntExist(tsym));
```

Comment:
I would suggest to use `.pos()`, instead of `.pos`. The former returns a `DiagnosticPosition`, with start/end/preferred positions, while the later just an integer. I.e. in JShell, the former will do:
```
jshell> import unknown.unknown.unknown.*;
|  Error:
|  package unknown.unknown.unknown does not exist
|  import unknown.unknown.unknown.*;
|         ^---------------------^
```

while the latter will do:
```
jshell> import unknown.unknown.unknown.*;
|  Error:
|  package unknown.unknown.unknown does not exist
|  import unknown.unknown.unknown.*;
|                        ^
```

Reply:
Fixed in [c01e851](https://github.com/openjdk/jdk/pull/19073/commits/c01e851fdfb95977a1898c7fabee119ee230ab83), thanks.